src/common/transformations/include/ov_ops/moe.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <array>
+#include <memory>
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/op.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov::op::internal {
+///
+/// \brief MOE experts
+class TRANSFORMATIONS_API MOE : public ov::op::Op {
+public:
+    OPENVINO_OP("MOE", "ie_internal_opset");
+
+    MOE() = default;
+
+    struct Config {
+        size_t topk = 0;
+        size_t expert_num = 0;
+        size_t hidden_size = 0;
+        size_t intermediate_size = 0;
+        size_t group_size = 0;  // quantized group size, 0 for no group size. same for gate/up/down
+        ov::element::Type weight_type = ov::element::dynamic;  // same for gate/up/down
+        ov::element::Type scale_type = ov::element::dynamic;   // same for gate/up/down
+        ov::element::Type zp_type = ov::element::dynamic;      // same for gate/up/down
```

Comment:
```suggestion
        size_t topk{};
        size_t expert_num{};
        size_t hidden_size{};
        size_t intermediate_size{};
        size_t group_size{};  // quantized group size, 0 for no group size. same for gate/up/down
        ov::element::Type weight_type{};  // same for gate/up/down
        ov::element::Type scale_type{};   // same for gate/up/down
        ov::element::Type zp_type{};      // same for gate/up/down
```
Optional and if works. It will use default initialization.